File: lib/lab_sound/gain_node.dart
import 'lab_sound.dart';
class GainNode extends AudioNode {
  late AudioParam gain;
  GainNode(AudioContext ctx): super(ctx, LabSound().createGain(ctx.pointer)) {
    this.gain = AudioParam(this.ctx, this.nodeId, LabSound().GainNode_gain(this.nodeId));
  }
}

File: bridging/GainNode.cpp
#include "dart_api/dart_api.h"
#include "LabSound/LabSound.h"
#include "KeepNode.cpp"
using namespace lab;

DART_EXPORT int createGain(AudioContext* context) {
    auto node = std::make_shared<GainNode>(*context);
    return keepNode(node);
}

DART_EXPORT int GainNode_gain(int nodeId) {
    auto node = std::static_pointer_cast<GainNode>(getNode(nodeId));
    return node ? keepAudioParam(nodeId, 0, node->gain()) : -1;
}
